sycl/test/kernel-and-program/kernel_cache.cpp

```diff
@@ -0,0 +1,148 @@
+// RUN: %clangxx -fsycl %s -o %t.out -lOpenCL
```

Comment:
I am not sure that you need to specify -lOpenCL flag in tests. Probably it would be better to get rid off all OpenCL function calls (like `clReleaseKernel`) and use `piKernelRelease` instead.

Reply:
I'm not sure it's legal to just call pi function without various macro. Do you suggest using PI_CALL(not sure what is the latest name) in test?

Reply:
I see the test uses clRelease* APIs only in order to decrement the reference counter which is incremented in the get() methods. We could just use internal method getHandlerRef which doesn't modify the reference counter instead.